tests/subsys/bluetooth/fast_pair/storage/account_key_storage/src/test_corrupted_data.c

```diff
@@ -79,24 +83,30 @@ static void self_test(void)
 	cu_account_keys_validate_loaded(ACCOUNT_KEY_MIN_ID, key_cnt);
 }
 
-static void self_test_rollover(void)
+ZTEST(suite_settings_mock_self_test, test_self_rollover)
 {
+	if ((ACCOUNT_KEY_CNT < 2) &&
+	    IS_ENABLED(CONFIG_TEST_BT_FAST_PAIR_STORAGE_OWNER_ACCOUNT_KEY)) {
+		ztest_test_skip();
+	}
+
 	int err;
 	uint8_t key_id = ACCOUNT_KEY_MIN_ID;
 	uint8_t key_cnt = 0;
 
 	for (size_t i = 0; i < ACCOUNT_KEY_CNT; i++) {
-		store_key(key_id);
+		if (i == 0) {
+			store_key(key_id, true);
+		} else {
+			store_key(key_id, false);
+		}
 		key_cnt++;
 		key_id = next_account_key_id(key_id);
 	}
 
-	store_key(key_id);
+	store_key(key_id, false);
```

Comment:
This would override Owner AK if Account Key count >= 2 in the configuration (not expected)